python/tskit/trees.py

```diff
@@ -4820,10 +4826,25 @@ def samples(self, population=None, population_id=None):
         if population_id is not None:
             population = population_id
         samples = self._ll_tree_sequence.get_samples()
+        keep = np.full(shape=samples.shape, fill_value=True)
         if population is not None:
             sample_population = self.tables.nodes.population[samples]
-            samples = samples[sample_population == population]
-        return samples
+            keep = np.logical_and(keep, sample_population == population)
+        if time is not None:
+            if isinstance(time, (int, float)):
```

Comment:
``if isinstance(time, numbers.Number)``

is more flexible here (current version will fail if you use a numpy 0D array as input)

Reply:
aha, thanks for letting me know!

Reply:
it doesn't look like this would work with a 0D array.

![image](https://user-images.githubusercontent.com/9950100/133804695-dcbd9d0c-6290-4dc3-8786-4c8fcc0089aa.png)